src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java

```diff
@@ -578,56 +576,48 @@ static String normalizeServer(String server) { /*package scope for testing*/
 
     @Restricted(NoExternalUse.class)
     public ApplicationContext createApplicationContext(LDAPSecurityRealm realm) {
-        DefaultInitialDirContextFactory initialDirContextFactory = new DefaultInitialDirContextFactory(getLDAPURL());
+        DefaultSpringSecurityContextSource contextSource = new DefaultSpringSecurityContextSource(getLDAPURL());
         if (getManagerDN() != null) {
-            initialDirContextFactory.setManagerDn(getManagerDN());
-            initialDirContextFactory.setManagerPassword(getManagerPassword());
+            contextSource.setUserDn(getManagerDN());
+            contextSource.setPassword(getManagerPassword());
         }
+        /* TODO no idea what to do with these
         Map<String, String> vars = new HashMap<>();
         vars.put(Context.REFERRAL, "follow");
         vars.put("com.sun.jndi.ldap.connect.timeout", "30000"); // timeout if no connection after 30 seconds
         vars.put("com.sun.jndi.ldap.read.timeout", "60000"); // timeout if no response after 60 seconds
         vars.putAll(getExtraEnvVars());
-        initialDirContextFactory.setExtraEnvVars(vars);
+        */
```

Comment:
Could not find any equivalent to this method in `DefaultInitialDirContextFactory`.

Anyway I think `extraEnvVars` is deprecated in the Jenkins plugin. Not sure about the JNDI properties.